### PR TITLE
#158 - Remove the printing of stack trace exception

### PR DESCRIPTION
**Pull request by mitch2na**

### CODE REVIEW

**Comment by remmeier** [comment]
could we work with log levels instead of removing exceptions?

WARN for exceptions with mappers and ERROR for exceptions without mapper seems appropriate. People then see the exceptions and can choose to ignore them if they like too.

I think the exception handling needs some work in general. People will have a hard time when only getting a response status/message.  And browsers currently only show a blank page in case of problems.

**Comment by mitch2na** [comment]
@remmeier I dont think adding a WARN level is appropriate if the developer has explicitly created an ExceptionMapper with the given ExceptionClass. This will only create more of a headache on the developer / operations if they have to tweak the logging levels to prevent stack traces from being printed by default. Seems like we are trying to adhere to a the case where a developer has not done there due diligence by logging an exception in the their mapper.

**Comment by sonarqube-reporter** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [KatharsisInvoker.java#L51](https://github.com/katharsis-project/katharsis-framework/blob/dc371743c1c805e4ae3bcb3416f8e490e2aea8c2/katharsis-servlet/src/main/java/io/katharsis/invoker/KatharsisInvoker.java#L51): Remove this unused "log" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. MAJOR [KatharsisFilterV2.java#L54](https://github.com/katharsis-project/katharsis-framework/blob/dc371743c1c805e4ae3bcb3416f8e490e2aea8c2/katharsis-spring/src/main/java/io/katharsis/spring/KatharsisFilterV2.java#L54): Remove this unused "log" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

File: katharsis-core/src/main/java/io/katharsis/errorhandling/mapper/KatharsisExceptionMapper.java [hunk 1]
```diff
@@ -2,14 +2,25 @@
...
 public final class KatharsisExceptionMapper implements JsonApiExceptionMapper<KatharsisMappableException> {
 
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
     @Override
     public ErrorResponse toErrorResponse(KatharsisMappableException exception) {
+        // log 5xx status as error and anything else as warn
+        if (exception.getHttpStatus() >= 500 || exception.getHttpStatus() < 600) {
+            logger.error("failed to process request", exception);
+        } else {
```
Comment (sonarqube-reporter) [inline katharsis-core/src/main/java/io/katharsis/errorhandling/mapper/KatharsisExceptionMapper.java]: MINOR This branch's code block is the same as the block for the branch on line 18. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1871)

**Comment by sonarqube-reporter** [comment]
SonarQube analysis reported 3 issues
* MAJOR 2 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [KatharsisInvoker.java#L51](https://github.com/katharsis-project/katharsis-framework/blob/54b85a8c29b60ce0e180c3be0a8f93811de87863/katharsis-servlet/src/main/java/io/katharsis/invoker/KatharsisInvoker.java#L51): Remove this unused "log" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)
1. MAJOR [KatharsisFilterV2.java#L54](https://github.com/katharsis-project/katharsis-framework/blob/54b85a8c29b60ce0e180c3be0a8f93811de87863/katharsis-spring/src/main/java/io/katharsis/spring/KatharsisFilterV2.java#L54): Remove this unused "log" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

**Comment by sonarqube-reporter** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [KatharsisInvoker.java#L33](https://github.com/katharsis-project/katharsis-framework/blob/ec66b92e55e41d1d9b683ab77e7f104b570e492f/katharsis-servlet/src/main/java/io/katharsis/invoker/KatharsisInvoker.java#L33): Remove this unused import 'org.slf4j.Logger'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [KatharsisInvoker.java#L34](https://github.com/katharsis-project/katharsis-framework/blob/ec66b92e55e41d1d9b683ab77e7f104b570e492f/katharsis-servlet/src/main/java/io/katharsis/invoker/KatharsisInvoker.java#L34): Remove this unused import 'org.slf4j.LoggerFactory'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [KatharsisFilterV2.java#L25](https://github.com/katharsis-project/katharsis-framework/blob/ec66b92e55e41d1d9b683ab77e7f104b570e492f/katharsis-spring/src/main/java/io/katharsis/spring/KatharsisFilterV2.java#L25): Remove this unused import 'org.slf4j.Logger'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [KatharsisFilterV2.java#L26](https://github.com/katharsis-project/katharsis-framework/blob/ec66b92e55e41d1d9b683ab77e7f104b570e492f/katharsis-spring/src/main/java/io/katharsis/spring/KatharsisFilterV2.java#L26): Remove this unused import 'org.slf4j.LoggerFactory'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by sonarqube-reporter** [comment]
SonarQube analysis reported 1 issue
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Pull request merged**